src/ucp/wireup/select.c

```diff
@@ -614,23 +731,28 @@ ucp_wireup_add_memaccess_lanes(ucp_wireup_select_ctx_t *select_ctx,
     snprintf(title, sizeof(title), criteria->title, "registered");
     mem_criteria.title           = title;
     mem_criteria.remote_md_flags = UCT_MD_FLAG_REG | criteria->remote_md_flags;
-    status = ucp_wireup_select_transport(ep, select_ctx->address_list,
-                                         select_ctx->address_count,
-                                         &mem_criteria, tl_bitmap, remote_md_map,
-                                         -1, -1, show_error, &select_info);
-    if (status != UCS_OK) {
+    status = ucp_wireup_select_lane(select_ctx, &mem_criteria, tl_bitmap,
+                                    remote_md_map, -1, -1, usage,
+                                    show_error, &select_info);
+    if (UCS_STATUS_IS_ERR(status)) {
         goto out;
     }
 
-    dst_md_index = select_ctx->address_list[select_info.addr_index].md_index;
-    reg_score    = select_info.score;
+    if (status == UCS_OK) {
+        dst_md_index = select_ctx->address_list[select_info.addr_index].md_index;
+        reg_score    = select_info.score;
 
-    /* Add to the list of lanes and remove all occurrences of the remote md
-     * from the address list, to avoid selecting the same remote md again. */
-    ucp_wireup_add_lane_desc(select_ctx, &select_info, dst_md_index, usage, 0);
-    remote_md_map &= ~UCS_BIT(dst_md_index);
-    tl_bitmap = ucp_wireup_unset_tl_by_md(ep, tl_bitmap,
-                                          select_info.rsc_index);
+        /* Add to the list of lanes and remove all occurrences of the remote md
+         * from the address list, to avoid selecting the same remote md again. */
+        ucp_wireup_add_lane(select_ctx, &select_info, usage);
+        remote_md_map &= ~UCS_BIT(dst_md_index);
+        tl_bitmap      = ucp_wireup_unset_tl_by_md(ep, tl_bitmap,
+                                                   select_info.rsc_index);
+    } else if (select_ctx->allow_am) {
+        ucs_assert(status == UCS_INPROGRESS);
+        /* Ask for emulation over AM needed for RMA/AMO lanes */
+        select_ctx->ep_init_flags |= UCP_EP_CREATE_AM_LANE;
```

Comment:
do we need to stop here?

Reply:
yes, we need to stop here - as we did it in the previous code